cylc/flow/scripts/cylc.py

```diff
@@ -663,3 +671,30 @@ def main():
                 if opts.version:
                     cmd_args.append("--version")
                 execute_cmd(command, *cmd_args)
+
+
+def handle_missing_dependency(
+    entry_point: pkg_resources.EntryPoint,
+    exc: ModuleNotFoundError
+) -> str:
+    """Return a suitable error message for a missing optional dependency.
+
+    Args:
+        entry_point: The entry point that was attempted to load but caused
+            a ModuleNotFoundError.
+        exc: The ModuleNotFoundError that was caught.
+
+    Re-raises the given ModuleNotFoundError if it is unexpected.
+    """
+    # If the missing module is an optional dependency, the EntryPoint
+    # should have an attr 'extras'
+    if not entry_point.extras:
+        raise exc
+
+    msg = f"cylc {entry_point.name}: missing optional dependency '{exc.name}'"
+    if entry_point.dist:
+        msg += "\nInstall using \"pip install '{0}[{1}]'\"".format(
+            entry_point.dist.project_name,
+            ','.join(entry_point.extras)
+        )
```

Comment:
Wondering if this is OTT

Reply:
I think it might be, somewhat confusing when installed via Conda, could give them a link to the installation section of the docs (although at present it doesn't really cover this).

Reply:
(This can happen for both Pip & Conda installs)

Reply:
- Created issue for documenting pip extras: https://github.com/cylc/cylc-doc/issues/430
- For Conda installs, wouldn't it only happen if they installed the `-base` recipe? E.g. `cylc-uiserver-base` instead of `cylc-uiserver`?

Reply:
> For Conda installs, wouldn't it only happen if they installed the -base recipe?

Yes.